src/Compositor.cpp

```diff
@@ -35,6 +35,14 @@ void handleUnrecoverableSignal(int sig) {
         return;
     }
 
+    // Kill the program if the crash-reporter is caught in a deadlock.
+    signal(SIGALRM, [](int _) {
+        char const* msg = "\nCrashReporter exceeded timeout, forcefully exiting\n";
+        write(2, msg, strlen(msg));
+        abort();
+    });
+    alarm(15);
```

Comment:
15 is a bit too long IMO, something like 10 should do

Reply:
In my testing it takes about 7 seconds to get the backtrace, and I'm on a fairly decent computer. Worse computers would probably exceed the 10 second threshold.

Reply:
hm, I wonder where that is spent, maybe in nm. Alright then